### ISSUE TITLE
Mutate data before form validation

### ISSUE THREAD

**Issue by vladsimon000000** [eighty9nine/filament-excel-import#20]
Or atleast make it easier to change the validator class

**Comment by eighty9nine** [comment]
Hello @acabust0000 , can you please provide more details perhaps with an example, so I understand!

**Comment by vladsimon000000** [comment]
We've had an issue with some of the excel versions auto-formatting field like phone numbers and dates. Even though we could set said fields as 'plain text' in the spreadsheets, plus symbols are still not displayed in correct format. We've worked around this by replacing the plus symbol with 2 zeros.

Once the data reaches the code, we've modifying it so it fits validation patterns such as [Propaganistas phone.](https://github.com/Propaganistas/Laravel-Phone)

It would've been great if the ValidationImport class would've be configurable (as I've seen other packages do) or at least allow for full configuration of Upload Field.

Were currently implementing data mutation using a junior-ish way:
```
$date = function (string $attribute, mixed $value, \Closure $fail) {
    try {
        if (! Carbon::make((string) str($value)->replace('.', '-'))->isAfter(today()->nextWeekday())) 
            $fail(__('validation.after');
    } catch (\Error) {
        $fail(__('validation.date');
    }
};

$phone = function (string $attribute, mixed $value, \Closure $fail) {
    if (! Rule::phone(str($value)->replaceFirst('00', '+'))) 
        $fail(__('validation.phone'));
};

ExcelImportAction::make()
  ->validateUsing([
    'delivery_date'         => ['required', $date],
    'expeditor_phone'       => ['required', $phone],
  ])
```

Thanks for the attention let me know if can help out with anything else!
Simon.

**Comment by eighty9nine** [comment]
I understand, I will add the mutate data before form validation method and add option to persist the mutation so the data is the same after validation. This is a great contribution!

**Comment by eighty9nine** [comment]
@acabust0000 the feature is now available in v3.1.2 you can find in the validation section of README
